pkg/endpoint/policy.go

```diff
@@ -682,7 +682,10 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 			regenerateTimeNs := time.Since(regenerateStart)
 			regenerateTimeSec := float64(regenerateTimeNs) / float64(time.Second)
 			e.getLogger().WithField(logfields.EndpointRegenerationTime, time.Since(regenerateStart).String()).Info("Regeneration of endpoint has completed")
-			metrics.EndpointRegenerationTime.Add(regenerateTimeSec)
+			metrics.EndpointRegenerationTime.With(map[string]string{
```

Comment:
This will be weird as if BPF compilation took place, we will account for the regeneration twice. Instead I think you have to:
1. Return a struct which contains the overall build time
2. Add labels depending on what operation was performed during regeneration:
   * BPF compilation
   * Policy evaluation
   * Proxy configuration
3. Have a single place where you update the metric and any combination of the above labels can be set.

That will allow us to see how long endpoint regenerations take and filter by label to see if only certain operations are subject to delays.